packages/ui/primitives/document-flow/add-fields.tsx

```diff
@@ -285,6 +296,15 @@ export const AddFieldsFormPartial = ({
     setSelectedSigner(recipients.find((r) => r.sendStatus !== SendStatus.SENT) ?? recipients[0]);
   }, [recipients]);
 
+  const recipientsByRole = recipients.reduce((acc, recipient) => {
```

Comment:
This looks cool from a code perspective, I wonder how it looks in the UI!

Also consider using a `groupBy` method, I think remeda might have it which we use in the `lib` package, I can't remember the details specifically since I was just trying my best to not use lodash 😆